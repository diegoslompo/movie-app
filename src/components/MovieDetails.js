import React, { Component } from 'react'
import PropTypes from 'prop-types'

class MovieDetails extends Component {

  static propTypes = {
    item: PropTypes.object
  }

  render () {

    const { item } = this.props

    return (
      <div className="ds-movie__details">
        <img src={item.Poster !== 'N/A' ? item.Poster : 'https://via.placeholder.com/180x270/000'} alt={item.Title}/>
        <h2>{item.Title}</h2>
        <p>Year: {item.Year}</p>
        <p>Type: {item.Type}</p>
        <p>Id: {item.imdbID}</p>
      </div>
    )
  }
} 
export default MovieDetails
