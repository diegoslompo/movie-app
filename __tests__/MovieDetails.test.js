import React from "react"
import ShallowRenderer from 'react-test-renderer/shallow';
import MovieDetails from '../src/components/MovieDetails'

describe('Testing component Movie Details', () => { 
    it('should render corretly', () => {
        const renderer = new ShallowRenderer()
        const wrapper = renderer.render(
            <MovieDetails item={{}} />
        )
        expect(wrapper).toMatchSnapshot()
    })
})