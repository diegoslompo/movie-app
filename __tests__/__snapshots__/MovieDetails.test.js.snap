// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing component Movie Details should render corretly 1`] = `
<div
  className="ds-movie__details"
>
  <img />
  <h2 />
  <p>
    Year: 
  </p>
  <p>
    Type: 
  </p>
  <p>
    Id: 
  </p>
</div>
`;
